src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -264,19 +264,29 @@ def _migrate_service_principals(
                     f"Service principal '{spn.permission_mapping.principal}' accesses storage account "
                     f"'{spn.permission_mapping.prefix}' with non-Allow network configuration"
                 )
+            try:
 
-            storage_credential_info = self._storage_credential_manager.create_with_client_secret(spn)
-            validation_results = self._storage_credential_manager.validate(
-                storage_credential_info,
-                spn.permission_mapping.prefix,
-            )
-            execution_result.append(validation_results)
+                storage_credential_info = self._storage_credential_manager.create_with_client_secret(spn)
+                validation_results = self._storage_credential_manager.validate(
+                    storage_credential_info,
+                    spn.permission_mapping.prefix,
+                )
+                execution_result.append(validation_results)
+            except (PermissionDenied, NotFound, BadRequest) as e:
+                logger.error(
+                    "Error creating storage credentials. Please check error message and try again."
```

Comment:
Similar update as above

Reply:
validation_results returns StorageCredentialValidationResult which contains the storage credential name and I use that to delete the storage credential on error